pkg/discovery/network/generic_test.go

```diff
@@ -120,30 +120,33 @@ var _ = Describe("Generic Network", func() {
 			Expect(clusterNet).NotTo(BeNil())
 		})
 
-		It("Should return the ClusterNetwork structure with pod CIDR", func() {
+		It("Should return the ClusterNetwork structure with all CIDR", func() {
 			Expect(clusterNet.PodCIDRs).To(Equal([]string{testPodCIDR}))
+			Expect(clusterNet.ServiceCIDRs).To(Equal([]string{testServiceCIDR}))
 		})
 
 		It("Should identify the networkplugin as generic", func() {
 			Expect(clusterNet.NetworkPlugin).To(BeIdenticalTo(cni.Generic))
 		})
-
-		It("Should return the ClusterNetwork structure with the service CIDR", func() {
-			Expect(clusterNet.ServiceCIDRs).To(Equal([]string{testServiceCIDRFromService}))
-		})
 	})
 
 	When("There is a kube-controller pod with the right parameter passed as an Arg", func() {
 		BeforeEach(func(ctx SpecContext) {
 			clusterNet = testDiscoverGenericWith(
 				ctx,
-				fakePodWithArg("kube-controller-manager", []string{"kube-controller-manager"}, "--cluster-cidr="+testPodCIDR),
+				fakePodWithArg("kube-controller-manager", []string{"kube-controller-manager"},
+					[]string{"--cluster-cidr=" + testPodCIDR, "--service-cluster-ip-range=" + testServiceCIDR}),
```

Comment:
Should be able to use:

```suggestion
				fakeKubeControllerManagerPod(),
```

Reply:
The test case is "There is a kube-controller pod with the right parameter **passed as an Arg**". The `fakeKubeControllerManagerPod` function should be and have been used in the last one naming "There is a kube-controller pod with the right parameter".